### ISSUE TITLE
Future fatalisation of vertical whitespace in headers will cause problems for pre-folded headers

### ISSUE THREAD

**Issue by cebjyre** [rjbs/Email-Simple#18]
**This is not yet a real problem**

The code introduced in 039b8c324b91af0b3654efe435787b2038b8158b appears to be preventing people from hurting themselves by either folding headers improperly or just having newlines etc in the headers for no real reason; this is fine. It is also detecting and warning about headers that (I believe) are properly folded in the first place (and then proceeds to discard that folding and refolds the header). At the moment this is just a warning so it isn't a real concern, but there is the threat of the code going fatal in the future, which I think should be limited to the case where the header contains newlines etc that aren't part of proper folding.

**Comment by pali** [comment]
> Future fatalisation of vertical whitespace in headers will cause problems for pre-folded headers

Just to note that already it is not possible to pass pre-folded headers into Email::Simple. You already found that commit, which replace any newline or vertical whitespace to normal space (+ throw warning about this fact).

Email::Simple fold and unfold long headers by its own...

So I do not think that future fatalisation would cause problems... but decision is of course on @rjbs.

**Comment by cebjyre** [comment]
No, the current state is that you **can** pass in pre-folded headers, but this code will unfold, and then refold if necessary.

```
$ perl -e 'use Email::MIME; use strict; use warnings; my $email = Email::MIME->create(header_str => [From => q{source@example.com}, To => q{dest@example.com}, Subject => ("x"x60 ."Decent\n Folding")], parts => [q[TEST!]]); warn $email->as_string;'

replaced vertical whitespace in Subject header with space; this will become fatal in a future version at -e line 1.
From: source@example.com
To: dest@example.com
Subject: xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxDecent
 Folding
Date: Mon, 18 Sep 2017 11:33:30 +1000
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
Content-Type: text/plain

TEST! at -e line 1.
```

If that warning were to become fatal, `$email` wouldn't have been created
